### ISSUE TITLE
julia unable to find compatible target in system image

### ISSUE THREAD

**Issue by hervyqa** [NixOS/nixpkgs#208700]
### Describe the bug
type Julia and an error message appears.

### Steps To Reproduce
1. add `julia` pkgs in configuration.nix
2. `$ sudo nixos-rebuild switch`
3. `$ julia`

```console
➜  ~ julia
ERROR: Unable to find compatible target in system image.
```

### Expected behavior
julia can run smoothly..

```console
$ julia

               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.8.4 (2022-12-23)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |
```

### Notify maintainers
@NickCao 

### Metadata

```console
➜  ~ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.85, NixOS, 22.11 (Raccoon), 22.11.1198.913a47cd064`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.1`
 - channels(hervyqa): `""`
 - channels(root): `"nixos-22.11"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by NickCao** [comment]
This should be fixed in https://github.com/NixOS/nixpkgs/pull/206327/commits/0fdda5326fc72e25aa69a706ea73db2574c3cfd9, would you like to test it on the nixos-unstable channel, and we can backport the fix if that's the case.

**Comment by hervyqa** [comment]
aaahh it's worked. thankyouu..

**Comment by hervyqa** [comment]
hi @NickCao, just curious. how to find out the status of binary julia has entered the 22.11 repo like in the unstable repo?

**Comment by NickCao** [comment]
You may check the latest build status at:
https://hydra.nixos.org/job/nixos/release-22.11/nixpkgs.julia_18.x86_64-linux
And find installation instructions  by clicking the links to individual builds.
(It unfortunately failed last time due to the flakiness of tests, hope we get better luck with the newly merged 1.8.5 release)

**Comment by hervyqa** [comment]
Thank you. for a while i used julia-bin.
